node/src/env0-deploy-utils.js

```diff
@@ -54,40 +52,89 @@ class DeployUtils {
   }
 
   async deployEnvironment(environment, blueprintRevision, blueprintId) {
-    console.log(`starting to deploy environmentId: ${environment.id}, blueprintId: ${blueprintId}`);
-    const deployment = await apiClient.callApi('post', `environments/${environment.id}/deployments`,
-        {data: {blueprintId, blueprintRevision}});
+    await this.waitForEnvironment(environment.id);
 
-    console.log(`Started deployment ${deployment.id}`);
+    return await apiClient.callApi('post', `environments/${environment.id}/deployments`, {data: {blueprintId, blueprintRevision}});
   }
 
   async destroyEnvironment(environment) {
-    console.log(`Starting to destroy environmentId: ${environment.id}`);
-    const deployment = await apiClient.callApi('post', `environments/${environment.id}/destroy`);
-    console.log(`Started destroy ${deployment.id}`);
+    await this.waitForEnvironment(environment.id);
+
+    return await apiClient.callApi('post', `environments/${environment.id}/destroy`);
+  }
+
+  async writeDeploymentStepLog(deploymentLogId, stepName) {
+    let hasMoreLogs;
+
+    do {
+      const { events, hasMoreLogs: currentHasMoreLogs } = await apiClient.callApi('get', `deployments/${deploymentLogId}/steps/${stepName}/log`);
+      events.forEach((event) => console.log(event.message));
+
+      hasMoreLogs = currentHasMoreLogs;
+      await apiClient.sleep(1000);
+    } while (hasMoreLogs)
   }
 
+  async fetchDeploymentSteps(deploymentLogId, stepsToSkip) {
```

Comment:
this function doesn't just `fetch`, can you give a more complete name please?

Reply:
changed it to `processDeploymentSteps`
91ac955